features/bsm/daemon/src/main/java/org/opennms/netmgt/bsm/daemon/Bsmd.java

```diff
@@ -97,6 +103,26 @@
 
     private boolean m_verifyReductionKeys = true;
 
+    private ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+
+    {
+        {
+            scheduledExecutorService.scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    if (reloadConfigurationAt > 0L && reloadConfigurationAt < System.currentTimeMillis()) {
+                        reloadConfigurationAt = 0L;
+                        final EventBuilder eventBuilder = new EventBuilder(EventConstants.RELOAD_DAEMON_CONFIG_UEI, "bsmd");
+                        eventBuilder.addParam(EventConstants.PARM_DAEMON_NAME, "bsmd");
+                        m_eventIpcManager.sendNow(eventBuilder.getEvent());
+                    }
+                }
+            }, RELOAD_DELAY, RELOAD_DELAY, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private long reloadConfigurationAt = 0L;
```

Comment:
I would mark this a volatile or use an AtomicLong - since it's written from one thread, and read/update by another